packages/core/src/actors.ts

```diff
@@ -274,48 +158,46 @@ export function createDeferredBehavior<TEvent extends EventObject>(
 
       return undefined;
     },
+    subscribe: (observer) => {
+      observers.add(observer);
+
+      return {
+        unsubscribe: () => {
+          observers.delete(observer);
+        }
+      };
+    },
     initialState: undefined
   };
 
   return behavior;
 }
 
-export function createPromiseBehavior<T, TEvent extends EventObject>(
+export function fromPromise<T, TEvent extends EventObject>(
   lazyPromise: Lazy<PromiseLike<T>>
 ): Behavior<{ type: string }, T | undefined> {
-  const parent = CapturedState.current?.actorRef;
   let canceled = false;
   const observers: Set<Observer<T>> = new Set();
 
   const behavior: Behavior<TEvent, T | undefined> = {
-    transition: (_, event, actorContext) => {
+    transition: (_, event) => {
       switch (event.type) {
         case startSignalType:
           const resolvedPromise = Promise.resolve(lazyPromise());
```

Comment:
just an observation for now - it's an interesting question if we should "adapt" this to a real promise through `Promise.resolve`, some other libraries don't do this: https://codesandbox.io/s/focused-brown-kojbgv?file=/src/index.js

The implementation in the `rxjs` can be found here: https://github.com/ReactiveX/rxjs/blob/47fa8d555754b18887baf15e22eb3dd91bf8bfea/src/internal/observable/innerFrom.ts#L84-L98